pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1072,6 +1093,82 @@ func (nc *Controller) handleDisruption(zoneToNodeConditions map[string][]*v1.Nod
 	}
 }
 
+func (nc *Controller) podUpdated(oldPod, newPod *v1.Pod) {
+	if len(newPod.Spec.NodeName) != 0 && (oldPod == nil || newPod.Spec.NodeName != oldPod.Spec.NodeName) {
+		nc.podUpdateQueue.Add(newPod)
+	}
+}
+
+func (nc *Controller) doPodProcessingWorker() {
+	for {
+		obj, shutdown := nc.podUpdateQueue.Get()
+		// "podUpdateQueue" will be shutdown when "stopCh" closed;
+		// we do not need to re-check "stopCh" again.
+		if shutdown {
+			return
+		}
+
+		pod := obj.(*v1.Pod)
+		nodeName := pod.Spec.NodeName
+
+		func() {
+			nc.nodeHealthLock.Lock()
+			defer nc.nodeHealthLock.Unlock()
+			defer nc.podUpdateQueue.Done(pod)
+
+			nodeHealth, exists := nc.nodeHealthMap[nodeName]
+			if !exists {
+				return
+			}
+
+			node, err := nc.nodeLister.Get(nodeName)
+			if err != nil {
+				klog.Errorf("Failed to list node %v.", nodeName)
+				return
+			}
+
+			gracePeriod := nc.nodeMonitorGracePeriod
+			_, currentReadyCondition := nodeutil.GetNodeCondition(nodeHealth.status, v1.NodeReady)
+			if currentReadyCondition == nil {
+				// Nothing to do. Pod will be handled by monitorNodeHealth when needed.
+				return
+			}
+
+			decisionTimestamp := nc.now()
+			if nc.useTaintBasedEvictions {
```

Comment:
It seems you're basically trying to replica this logic:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L727-L802

So this seems incorrect - it seems it should be if "!" nc.use...

That said - this is generating a lot of copy-pasted code, which is very bad.